src/vsphere_cpi/lib/cloud/vsphere/vcenter_client.rb

```diff
@@ -482,8 +482,22 @@ def find_child_by_name(mob, child_path)
       find_child_by_name(mob.child_entity.find {|c| c.name == child_entity_name }, child_path)
     end
 
+    def dvpg_istype_nsxt?(key:, dc:)
+      logger.info("Checking if #{key} is backed by NSXT")
+      portgroup = dc.mob.network.detect do |network|
```

Comment:
This looks initially good but we have to make this logic more complicated to account for the fact that the relevant portgroup could be nested inside something else. As far as I know this only looks at the direct children of the datacenter's `networkFolder`.

Reply:
I think it recursively checks all children of networkFolder. (verified using VC's mob browser). @EleanorRigby - can you please confirm if this is right?

Reply:
I think Network will list all possible Networks in Datacenter. And using networkFolder to recursively gather should bring no difference.

Reply:
We already use this in find_network to reliably get a list of all networks available under the Datacenter.